src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNodeVector.cs

```diff
@@ -0,0 +1,168 @@
+			}
+
+			return parentItemsSource;
+		}
+
+		public void Append(TreeViewNode item, bool updateItemsSource = true)
+		{
+			InsertAt(Count, item, updateItemsSource);
+		}
+
+		public override void Add(TreeViewNode item) => Append(item);
+
+		public void InsertAt(int index, TreeViewNode item, bool updateItemsSource = true)
+		{
+			var inner = GetVectorInnerImpl();
+			if (m_parent == null)
+			{
+				throw new InvalidOperationException("Parent node must be set");
+			}
+			if (index > inner.Count)
+			{
+				throw new IndexOutOfRangeException("Index out of range for Insert");
+			}
+
+			item.Parent = m_parent;
+
+			inner.Insert(index, item);
+
+			if (updateItemsSource)
+			{
+				var itemsSource = GetWritableParentItemsSource();
+				if (itemsSource != null)
+				{
+					itemsSource.Insert(index, item.Content);
+				}
+			}
+		}
+
+		public override void Insert(int index, TreeViewNode item) => InsertAt(index, item);
+
+		public void SetAt(int index, TreeViewNode item, bool updateItemsSource = true)
+		{
+			RemoveAt(index, updateItemsSource);
+			InsertAt(index, item, updateItemsSource);
+		}
+		
+		public override TreeViewNode this[int index]
+		{
+			get => base[index];
+			set => SetAt(index, value);
+		}
+
+		public void RemoveAt(int index, bool updateItemsSource = true)
+		{
+			var inner = GetVectorInnerImpl();
+			var targetNode = inner[index];
+			targetNode.Parent = null;
+
+			inner.RemoveAt(index);
+
+			if (updateItemsSource)
+			{
+				var source = GetWritableParentItemsSource();
+				if (source != null)
+				{
+					source.RemoveAt(index);
+				}
+			}
+		}
+
+		public override void RemoveAt(int index) => RemoveAt(index, true);		
+
+		public void RemoveAtEnd(bool updateItemsSource = true)
+		{
+			var index = GetVectorInnerImpl().Count - 1;
+			RemoveAt(index, updateItemsSource);
+		}
+
+		public void ReplaceAll(TreeViewNode[] values, bool updateItemsSource = true)
+		{
+			var inner = GetVectorInnerImpl();
+
+			var count = inner.Count;
+			if (count > 0)
+			{
+				Clear(updateItemsSource);
+
+				var itemsSource = GetWritableParentItemsSource();
+				// Set parent on new elements
+				if (m_parent == null)
+				{
+					throw new InvalidOperationException("Parent must be set");
+				}
+				foreach (var value in values)
+				{
+					value.Parent = m_parent;
+					if (itemsSource != null)
+					{
+						itemsSource.Add(value.Content);
+					}
+				}
+
+				inner.Clear();
+				inner.AddRange(values);
+			}
+		}
+
+		public void Clear(bool updateItemsSource = true)
+		{
+			var inner = GetVectorInnerImpl();
+			var count = inner.Count;
+
+			if (count > 0)
+			{
+				for (var i = 0; i < count; i++)
+				{
+					var node = inner[i];
+					node.Parent = null;
+				}
+
+				inner.Clear();
+
+				if (updateItemsSource)
+				{
+					var itemsSource = GetWritableParentItemsSource();
+					if (itemsSource != null)
+					{
+						itemsSource.Clear();
+					}
+				}
+			}
+		}
+
+		public override void Clear() => Clear(true);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this default value assigned to parameter 'updateItemsSource'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)